charts/images.yaml

```diff
@@ -111,8 +111,8 @@ images:
   tag: v0.56.2
 - name: kube-state-metrics
   sourceRepository: github.com/kubernetes/kube-state-metrics
-  repository: quay.io/coreos/kube-state-metrics
-  tag: v1.9.7
+  repository: registry.k8s.io/kube-state-metrics/kube-state-metrics
+  tag: "v2.5.0"
```

Comment:
I don't see compatibility promises for v2.5.0 for K8s < 1.20 - ref https://github.com/kubernetes/kube-state-metrics/tree/v2.5.0#compatibility-matrix. Shouldn't we continue to use v1.9.7 for K8s < 1.20?

Reply:
The problem is that that old version does not support the current k8s api server version. More specifically at one point we saw that it couldn't reconnect to the api server with this error message:
<img width="300" alt="image" src="https://user-images.githubusercontent.com/23032437/176191614-c776267a-1e19-40f3-ba3d-8adb0189820e.png">

Does the support requirement mean that we need to deploy a "matching" version of kube-state-metrics?

Note that the compatibility matrix in the kube-state-metrics readme is only showing the last 5 entries, and if we check the older versions, e.g. https://github.com/kubernetes/kube-state-metrics/tree/v2.1.1, there seems to be no backwards incompatible entries for those. So we could assume that the current version 2.5.0 is compatible with k8s 1.17+, and the table is just truncated to show only 5 entries. cc @wyb1

Reply:
> The problem is that that old version does not support the current k8s api server version. More specifically at one point we saw that it couldn't reconnect to the api server with this error message:

You should be able to use something like:
```yaml
- name: kube-state-metrics
  sourceRepository: github.com/kubernetes/kube-state-metrics
  repository: quay.io/coreos/kube-state-metrics
  tag: v1.9.7
  targetVersion: "< 1.20"
- name: kube-state-metrics
  sourceRepository: github.com/kubernetes/kube-state-metrics
  repository: registry.k8s.io/kube-state-metrics/kube-state-metrics
  tag: v2.5.0
  targetVersion: ">= 1.20"
```

Reply:
👍 Applied the change in a new commit and will verify it in the local setup.

Reply:
Using the local setup (https://github.com/gardener/gardener/blob/master/docs/deployment/getting_started_locally.md) I created a shoot with k8s version 1.19.0 and could confirm that the kube-state-metrics image v2.1.1 was used, i.e. the image version selection based on the targeted k8s cluster version worked as expected.

I could also verify that this older version of kube-state-metrics, which has compatibility matrix entries for k8s versions 1.17-1.21 (https://github.com/kubernetes/kube-state-metrics/tree/v2.1.1), is exposing the new metric names (e.g. `kube_node_status_allocatable` instead of `kube_node_status_allocatable_cpu_cores`) so that the adjusted dashboards and recording rules are working with this older version.

> Compatibility matrix
> At most, 5 kube-state-metrics and 5 [kubernetes releases](https://github.com/kubernetes/kubernetes/releases) will be recorded below.

Note that the compatibility matrix is limited in size to 5 entries, so maybe we could use a single version to cover all the supported k8s versions (https://github.com/gardener/gardener#k8s-conformance-test-coverage-). With this change, we are using 2 kube-state-metrics versions to cover the full range of k8s versions (1.17-1.24) inline with its documentation.